### PR TITLE
schema change for parameter set, parameter set and pipeline to refere…

### PR DESCRIPTION
**Pull request by cindyzh-ibm**
…nce parameter set

 
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: common-pipeline/pipeline-flow/pipeline-flow-v3-schema.json [hunk 1]
```diff
@@ -639,6 +639,31 @@
...
+      "properties": {
+        "name": {
+          "description": "The disambiguating name of the parameter set.",
+          "type": "string"
+        },
+        "ref": {
+          "description": "A reference to a parameter set by ID.",
+          "type": "string"
+        },
+        "catalog_ref": {
```
Comment (matthoward366) [inline common-pipeline/pipeline-flow/pipeline-flow-v3-schema.json:655]: Do we also need space_ref?

File: common-pipeline/parameters/parameters-v3-schema.json [hunk 2]
```diff
@@ -33,95 +35,42 @@
...
-            "type": {
+            "type_code": {
                "description": "The parameter type.",
                "enum": [
                   "string",
-                  "integer",
-                  "object",
-                  "array",
-                  "boolean"
+                  "encrypted",
```
Comment (matthoward366) [inline common-pipeline/parameters/parameters-v3-schema.json:42]: Where did these types come from?  These don't match the types of other data types in the pipeline flow schemas.  If this needs to be open then we shouldn't use an enum.

Comment (cindyzh-ibm) [reply common-pipeline/parameters/parameters-v3-schema.json:42]: @matthoward366 
Matt, 
Here is the list of parameter type for both local parameter and parameter in parameter set.

image

Why the parameter type has to match the ones used in pipeline flow ?

Comment (matthoward366) [reply common-pipeline/parameters/parameters-v3-schema.json:42]: Why does this have to be an enum even?  Remember this is for anyone to use not just your product.

Comment (matthoward366) [reply common-pipeline/parameters/parameters-v3-schema.json:42]: On top of that why not keep this as `type`.  That match `subtype` below better.

Comment (cindyzh-ibm) [reply common-pipeline/parameters/parameters-v3-schema.json]: @matthoward366 
Matt,
Changes are made. Please review again. Thanks

**Pull request merged**